### PR TITLE
Issue-612 Fixed typos all over the codebase. Tests still pass.

### PR DESCRIPTION
**Pull request by theodesp**
Signed-off-by: Theo <theo.despoudis@teckro.com>

## Description

I went through all the code base and corrected all the logical typos.
## Related Issue

https://github.com/inversify/InversifyJS/issues/612

## Motivation and Context

Fixed typos over the code base
## How Has This Been Tested?

Run the tests after all the typo fixes.
## Types of changes

- [x] Updated docs / Refactor code / Added a tests case (non-breaking change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/planning/planner.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { Request } from "./request";
 import { Target } from "./target";
 import { BindingCount } from "../bindings/binding_count";
 import { getDependencies } from "./reflection_utils";
-import { Metadata } from "../planning/metadata";
+import { Metadata } from "./metadata";
```
Comment (theodesp) [inline src/planning/planner.ts:8]: Simplified path

File: src/planning/target.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import { interfaces } from "../interfaces/interfaces";
-import { Metadata } from "../planning/metadata";
+import { Metadata } from "./metadata";
```
Comment (theodesp) [inline src/planning/target.ts:2]: Simplified path

File: src/inversify.ts [hunk 3]
```diff
@@ -13,5 +13,5 @@ export { MetadataReader } from "./planning/metadata_reader";
 export { guid } from "./utils/guid";
 export { interfaces } from "./interfaces/interfaces";
 export { decorate } from "./annotation/decorator_utils";
-export { traverseAncerstors, taggedConstraint, namedConstraint, typeConstraint } from "./syntax/constraint_helpers";
+export { traverseAncestors, taggedConstraint, namedConstraint, typeConstraint } from "./syntax/constraint_helpers";
```
Comment (lholznagel) [inline src/inversify.ts]: Just as an FYI: This is a possible breaking change

Comment (theodesp) [reply src/inversify.ts]: Ok I can revert that

**Comment by remojansen** [comment]
Hi @theodesp thanks a lot for this PR :+1: great work once more :smile:

**Pull request merged**